programs/host/frontend.c

```diff
@@ -0,0 +1,935 @@
+/*
+ *  Host for offloaded functions
+ *
+ *  This program receives serialized function calls (see library/serialize.c)
+ *  and executes them.
+ *
+ *  This program currently requires the serialization channel to be on file
+ *  descriptors 3 for target-to-host and 4 for host-to-target.
+ *  Set the environment variable FRONTEND_DEBUG to get debugging traces.
+ *
+ *  See library/serialize.c for a description of the serialization format.
+ *
+ *  Copyright (C) 2017, ARM Limited, All Rights Reserved
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  This file is part of mbed TLS (https://tls.mbed.org)
+ */
+
+#include "frontend-config.h"
+
+#include <stdint.h>
+#include <errno.h>
+
+#if defined(MBEDTLS_PLATFORM_C)
+#include "mbedtls/platform.h"
```

Comment:
`mbedtls/platform.h` includes `config.h`, which as I understood is deliberately omitted here and replaced by `frontend-config.h`.

Reply:
`mbedtls/platform.h` also includes `frontend-config.h` via macro `MBEDTLS_CONFIG_FILE` set in the Makefile. See
https://github.com/ARMmbed/mbedtls/blob/782893cb78de186ac3fe44a19828c3c2d776e6a5/programs/Makefile#L119

`frontend-config.h` is included directly here to avoid impression that it can be compiled with any config.

Reply:
Ah, thanks for explaining that!